### DISCUSSION TITLE
Prevent /up healthchecks from appearing in access logs

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/docker-php#409] [Ideas & Feature Requests]
# 👉 Describe the problem
[comment]: <> (What are you trying to solve?)
- If someone uses the Laravel `/up` health check, the access logs get noisy with every check.
- Only the default `/healthcheck` supports ignores sending requests to the access log

# 👥 Problem evidence & reach
[comment]: <> (How many people have this problem?)
https://x.com/angusmcritchie/status/1818550305363230764

# 🏆 Proposed Solution
[comment]: <> (Describe the feature that you are proposing and how it will solve the problem)
There are two options I see as of now

### "Synchronize" The `HEALTHCHECK_PATH` variable
- `HEALTHCHECK_PATH` could set all the server configs for NGINX, Apache, Unit, etc
- With this path, it will be in a block that ignores routing to the access log

### Add a second block to ignore `/up` as well
If the above solution becomes to complicated to support, then it is possible to just ignore `/up`. The only problem is that makes a pretty big assumption that no one will use the `/up` endpoint in their application.

# 🥰 Describe the "impact" on users?
[comment]: <> (How will this make people's lives better once it is solved?)
This will reduce noisy logs on production machines for people using the Laravel healthcheck endpoint.

# 💯 How do we validate the problem is solved?
[comment]: <> (Explain use cases on how we can measure the success of this implementation)
- [ ] Health checks continue to work
- [ ] Laravel healtchecks to `/up` do not appear in logs

**Comment by angus-mcritchie** [comment]
[Server Side Up provides a way to add custom startup scripts](https://serversideup.net/open-source/docker-php/docs/customizing-the-image/adding-your-own-start-up-scripts), so I created my own to ignore Laravel's `/up` health checks, here's how I did it.

1. Add a `.docker/entrypoint.d/15-suppress-laravel-up-logs.sh` file to your project.
2. Add `COPY --chmod=755 .docker/entrypoint.d/ /etc/entrypoint.d/` to your `Dockerfile`
3. Add `ENV SUPPRESS_LARAVEL_UP_LOGS=true` to your `Dockerfile`

Contents of `.docker/entrypoint.d/15-suppress-laravel-up-logs.sh`
```bash
#!/command/with-contenv sh

if [ "$SUPPRESS_LARAVEL_UP_LOGS" = "true" ]; then
    echo "ℹ️  Suppressing Laravel's /up access logs."

    echo '' >> /etc/apache2/vhost-templates/https.conf
    echo '# Suppress Laravel /up endpoint logs' >> /etc/apache2/vhost-templates/https.conf
    echo 'SetEnvIf Request_URI "^/up$" dontlog' >> /etc/apache2/vhost-templates/https.conf

    echo '' >> /etc/apache2/vhost-templates/http.conf
    echo '# Suppress Laravel /up endpoint logs' >> /etc/apache2/vhost-templates/http.conf
    echo 'SetEnvIf Request_URI "^/up$" dontlog' >> /etc/apache2/vhost-templates/http.conf
else
    echo "ℹ️ Suppressing Laravel's /up access logs is not enabled, enable it by setting SUPPRESS_LARAVEL_UP_LOGS=true."
fi
```

Nice and clean logs now.

Happy coding 🙏

**Reply by jaydrogers** [reply]
I just added native support for this here: https://github.com/serversideup/docker-php/pull/283/commits/1017afa47c878b54ef44efa0437776d07a1a8d2c

I had to put an extra filter in there:

<img width="513" height="70" alt="image" src="https://github.com/user-attachments/assets/c48d3f7f-dae3-443c-a51f-15383bde95fb" />

This should be available in v4.0: https://github.com/serversideup/docker-php/pull/283

No custom script needed 😃
